src/App.tsx

```diff
@@ -1,26 +1,131 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import * as todosService from './api/todos';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Todo } from './types/Todo';
+import { Error } from './components/Error/Error';
+import { ErrorEnum } from './types/ErrorEnum';
+import { CompleteStatus } from './types/CompleteStatus.enum';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorEnum | null>(null);
+  const [filter, setFilter] = useState(CompleteStatus.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const getTodosHandler = useCallback(() => {
+    todosService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage(ErrorEnum.LOAD));
+  }, []);
+
+  useEffect(() => {
+    getTodosHandler();
+  }, [getTodosHandler]);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const addTodoHandler = useCallback(() => {
+    if (query.trim() === '') {
+      setErrorMessage(ErrorEnum.TITLE);
+
+      return;
+    }
+
+    const newTodo = {
+      userId: todosService.USER_ID,
+      title: query.trim(),
+      completed: false,
+    };
+
+    setIsLoading(true);
+    setTempTodo({ id: 0, ...newTodo });
```

Comment:
```suggestion
    setTempTodo({ id: 0, ...newTodo });
```
wiil be great to Omit id for temp todo in this case you shouldn't write this kostyl with id ;)

Reply:
It was a condition of this task.